pennylane_lightning/core/src/simulators/lightning_qubit/gates/GateIndices.cpp

```diff
@@ -47,4 +48,97 @@ auto generateBitPatterns(const std::vector<std::size_t> &qubitIndices,
     }
     return indices;
 }
+
+auto generatePauliWordIndices(const std::string &word)
+    -> std::vector<std::size_t> {
+    const std::size_t n_word = word.size();
+    const std::size_t n_indices = Pennylane::Util::exp2(n_word);
+    std::vector<std::size_t> indices(n_indices);
+    std::vector<std::size_t> mtype(n_word);
+    std::transform(word.begin(), word.end(), mtype.begin(), [](auto ch) {
+        return (ch == 'I' || ch == 'Z') ? 0UL : 1UL;
+    });
+    std::size_t current_size = 2;
+    indices[0] = 0UL;
+    indices[1] = 1UL;
+    if (mtype[n_word - 1 - 0] == 1) {
```

Comment:
```suggestion
    if (mtype[n_word - 1] == 1) {
```
Is it necessary the `- 0` ?